cmd/package_managers.go

```diff
@@ -77,6 +89,61 @@ type rubyGemsSearchResults struct {
 	SourceCodeURI string `json:"source_code_uri"`
 }
 
+type nugetIndexResult struct {
+	ID   string `json:"@id"`
+	Type string `json:"@type"`
+}
+
+type nugetIndexResults struct {
+	Resources []nugetIndexResult `json:"resources"`
+}
+
+type nugetPackageRegistrationCatalogRoot struct {
+	Pages []nugetPackageRegistrationCatalogPage `json:"items"`
+}
+
+type nugetPackageRegistrationCatalogPage struct {
+	ID       string                            `json:"@id"`
+	Packages []nugetPackageRegistrationPackage `json:"items"`
+}
+
+type nugetPackageRegistrationPackage struct {
+	Entry nugetPackageRegistrationCatalogEntry `json:"catalogEntry"`
+}
+
+type nugetPackageRegistrationCatalogEntry struct {
+	Version string `json:"version"`
+	Listed  bool   `json:"listed"`
+}
+
+func (entry *nugetPackageRegistrationCatalogEntry) UnmarshalJSON(text []byte) error {
+	type Alias nugetPackageRegistrationCatalogEntry
+	aux := Alias{
```

Comment:
Is the type alias needed?

Reply:
it is, otherwise the UnmarshalJSON method goes into infinite loop.
see here: https://stackoverflow.com/questions/30445479/how-to-specify-default-values-when-parsing-json-in-go 
and here: https://stackoverflow.com/questions/59190078/how-to-set-a-default-value-when-unmarshal-a-json-to-struct